full/full_coupler_mod.F90

```diff
@@ -1694,19 +1693,80 @@ subroutine coupler_flux_check_stocks(nc, Time, Atm, Land, Ice, Ocean_state, coup
 
     integer, intent(in) :: nc                       !< current outerloop timestep
     type(FmsTime_type), intent(in) :: Time          !< Time
-    type(atmos_data_type), intent(inout) :: Atm     !< Atm 
+    type(atmos_data_type), intent(inout) :: Atm     !< Atm
     type(land_data_type), intent(inout)  :: Land    !< Land
     type(ice_data_type), intent(inout)   :: Ice     !< Ice
     type(ocean_state_type), pointer, intent(inout) :: Ocean_state    !< Ocean_state
     type(coupler_clock_type), intent(inout)        :: coupler_clocks !< coupler_clocks
-    
+
     call fms_mpp_clock_begin(coupler_clocks%flux_check_stocks)
-    if (check_stocks*((nc-1)/check_stocks) == nc-1 .AND. nc > 1) then      
+    if (check_stocks*((nc-1)/check_stocks) == nc-1 .AND. nc > 1) then
       call fms_mpp_set_current_pelist()
       call flux_check_stocks(Time=Time, Atm=Atm, Lnd=Land, Ice=Ice, Ocn_state=Ocean_state)
     endif
     call fms_mpp_clock_end(coupler_clocks%flux_check_stocks)
-    
+
   end subroutine coupler_flux_check_stocks
-  
+
+  !> \brief This subroutine calls flux_ocean_to_ice
+  subroutine coupler_flux_ocean_to_ice(Ocean, Ice, Ocean_ice_boundary, coupler_clocks, slow_ice_ocean_pelist)
+
+    implicit none
+
+    type(ocean_public_type), intent(inout) :: Ocean  !< Ocean
+    type(ice_data_type),     intent(in)    :: Ice    !< Ice
+    type(ocean_ice_boundary_type), intent(inout) :: Ocean_ice_boundary !< Ocean_ice_boundary
+    type(coupler_clock_type), intent(inout) :: coupler_clocks          !< coupler_clocks
+    integer, dimension(:),    intent(in)    :: slow_ice_ocean_pelist   !< slow_ice_ocean_pelist
+
+    !Redistribute quantities from Ocean to Ocean_ice_boundary
+
+    ! If the slow ice is on a subset of the ocean PEs, use the ocean PElist.
+    call fms_mpp_set_current_pelist(slow_ice_ocean_pelist)
+    call fms_mpp_clock_begin(coupler_clocks%flux_ocean_to_ice)
+
+    !Ice intent is In, used only for accessing Ice%area and knowing if we are on an Ice pe
+    call flux_ocean_to_ice(Ocean, Ice, Ocean_ice_boundary)
+
+    call fms_mpp_clock_end(coupler_clocks%flux_ocean_to_ice)
+
+  end subroutine coupler_flux_ocean_to_ice
+
+!> \brief This subroutine calls flux_ocean_to_ice
```

Comment:
This is also too brief.  This routine sets clocks and does a check before it calls the routine

Reply:
Added.